src/bindings/c/include/openvino/ov_model.h

```diff
@@ -0,0 +1,145 @@
+ */
+
+#pragma once
+
+#include "ov_common.h"
+#include "ov_node.h"
+#include "ov_partial_shape.h"
+
+typedef struct ov_model ov_model_t;
+
+// Model
+/**
+ * @defgroup model model
+ * @ingroup openvino_c
+ * Set of functions representing of Model and Node.
+ * @{
+ */
+
+/**
+ * @brief Release the memory allocated by ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t to free memory.
+ */
+OPENVINO_C_API(void) ov_model_free(ov_model_t* model);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param output_nodes A pointer to the ov_output_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_outputs(const ov_model_t* model, ov_output_node_list_t* output_nodes);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param input_nodes A pointer to the ov_input_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_inputs(const ov_model_t* model, ov_output_node_list_t* input_nodes);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_name input tensor name (char *).
+ * @param input_node A pointer to the ov_output_const_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_input_by_name(const ov_model_t* model, const char* tensor_name, ov_output_const_node_t** input_node);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param index input tensor index.
+ * @param input_node A pointer to the ov_input_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_input_by_index(const ov_model_t* model, const size_t index, ov_output_const_node_t** input_node);
+
+/**
+ * @brief Returns true if any of the op's defined in the model contains partial shape.
+ * @param model A pointer to the ov_model_t.
+ */
+OPENVINO_C_API(bool) ov_model_is_dynamic(const ov_model_t* model);
+
+/**
+ * @brief Do reshape in model with partial shape for a specified name.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_name input tensor name (char *).
+ * @param partialShape A PartialShape.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape_by_name(const ov_model_t* model, const char* tensor_name, const ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Do reshape in model with a list of <name, partial shape>.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_names input tensor name (char *) list.
+ * @param partialShape A PartialShape list.
+ * @param cnt The item count in the list.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape_by_names(const ov_model_t* model,
```

Comment:
For port we have only one function with array. Can we also keep only one method? Or will be better to add new method for port?

Reply:
The C++ interface for port reshape is:

     void reshape(const std::map<size_t, ov::PartialShape>& partial_shapes);

The C interface is defined as:

      ov_status_e ov_model_reshape_by_ports(const ov_model_t* model, size_t* ports, const ov_partial_shape_t** artial_shape, size_t cnt);

Reply:
What does " keep only one method" mean?